build-system/tasks/cherry-pick.js

```diff
@@ -103,65 +103,67 @@ function performCherryPick(sha) {
 }
 
 function cherryPick() {
-  const {push, remote = 'origin'} = argv;
-  const commits = (argv.commits || '').split(',').filter(Boolean);
-  let onto = String(argv.onto || '');
+  return new Promise((resolve, reject) => {
```

Comment:
Please make this whole function async, then we no longer need the Promise constructor or any of the explicit resolve/reject calls and can make use of normal return/throw

Reply:
Hmm, I'd rather not wrap git commands in async functions.

Reply:
Related: https://github.com/ampproject/amphtml/issues/26562

Reply:
> Hmm, I'd rather not wrap git commands in async functions.
Can you clarify what you mean by that? Wrapping it in a promise constructor is the same as wrapping in an async function

Reply:
Adding `async` to the function declaration, removing the Promise constructor, and replacing `reject` with `throw` and `resolve` with `return` results in an identical function, just using the updated syntax

Reply:
That issue seems unrelated? This whole function is synchronous because it's just git commands. There isn't a point in "making" it async.

Reply:
Nevermind, one sec

Reply:
A function returning a promise is strictly equal to an async function. I believe this is how almost all files under `build-system/tasks` are written now